pkgs/build-support/node/fetch-npm-deps/default.nix

```diff
@@ -0,0 +1,52 @@
+{ lib, stdenv, rustPlatform, Security }:
+
+rec {
+  fetch-npm-deps = rustPlatform.buildRustPackage {
+    pname = "fetch-npm-deps";
+    version = (lib.importTOML ./Cargo.toml).package.version;
+
+    src = ./.;
+
+    cargoLock.lockFile = ./Cargo.lock;
+
+    buildInputs = lib.optional stdenv.isDarwin Security;
+  };
+
+  fetchNpmDeps =
+    { name ? "npm-deps"
+    , src ? null
+    , srcs ? [ ]
+    , patches ? [ ]
+    , sourceRoot ? [ ]
+    , hash ? ""
+    , ...
+    } @ args: stdenv.mkDerivation ({
```

Comment:
I'm not sure why these are set to their default/empty values in [`fetchCargoTarball`](https://github.com/NixOS/nixpkgs/blob/3ac8876ca335877e9d557aec2d8dd1923ee5bbab/pkgs/build-support/rust/fetch-cargo-tarball/default.nix#L20-L28)... I did it here anyways, though I'm guessing they can probably be removed.

Reply:
I think `{ foo ? {} } @ args:` doesn't keep the `{}` for `foo`, `args` removes the default.